packages/perseus-editor/src/util/modernize-widgets-utils.ts

```diff
@@ -0,0 +1,205 @@
+import type {
+    NumericInputWidget,
+    PerseusRenderer,
+    PerseusWidgetsMap,
+} from "@khanacademy/perseus";
+
+type WidgetRenameMap = {
+    [oldKey: string]: string;
+};
+
+// This utils in this file are used to modernize our Perseus JSON structure,
+// so that we can convert deprecated widgets to their modern equivalents when
+// content creators use the Editor Page to update content containing these widgets.
+// Currently, we're only converting input-number to numeric-input,
+// but we can add more conversions here in the future.
+// Modernize the json content of a PerseusRenderer object
+// by converting deprecated widgets to their modern equivalents
+export const convertDeprecatedWidgets = (
+    json: PerseusRenderer,
+): PerseusRenderer => {
+    // If there's no json, or no input-number widgets, then we can return the json as is
+    if (!json || !conversionRequired(json)) {
+        return json;
+    }
+
+    // Currently we're only converting input-number to numeric-input,
+    // But we can add more conversions here in the future
+    const modernJson = inputNumberToNumericInput(json);
+
+    return modernJson;
+};
+
+const conversionRequired = (json: PerseusRenderer): boolean => {
+    const inputNumberRegex = /input-number \d+/; // We can add more regexes here in the future
+
+    // Check the content string for any top-level input-number widgets
+    if (inputNumberRegex.test(json.content)) {
+        return true;
+    }
+
+    // If there's no input-numbers top-level, then check the widgets for any nested widgets
+    for (const key of Object.keys(json.widgets)) {
+        if (json.widgets[key].options.widgets) {
+            const nestedJson = json.widgets[key].options;
+            if (conversionRequired(nestedJson)) {
+                return true;
+            }
+        }
+    }
+
+    return false;
+};
+
+// Convert the deprecated widget refs in the content string
+// of a PerseusRenderer object to their renamed equivalents
+const convertDeprecatedWidgetsInContent = (
+    json: PerseusRenderer,
+    renameMap: WidgetRenameMap,
+): string => {
+    return Object.keys(renameMap).reduce((newContent, oldKey) => {
+        const newKey = renameMap[oldKey];
+        if (newKey) {
+            return newContent.replace(oldKey, newKey);
+        }
+        return newContent;
+    }, json.content);
+};
+
+// Methods to convert input-number widgets to numeric-input widgets
+
```

Comment:
What do you think about pulling out widget-specific conversion functions to their own files (i.e. `util/deprecations/input-number.ts`)? This would help localize conversion logic, and would prevent bloat in this file as we add more deprecated widgets.